### PR TITLE
Add Spectron test framework for OpenFin and Electron

### PR DESCRIPTION
**Pull request by BenLambertNcl**
Adds Spectron testing framework for OpenFin and Electron. Tests do not currently run on CI, so a separate PR once that is figured out. Browser tests will also be added in another PR.

### CODE REVIEW

File: packages/api-electron-demo/package.json [hunk 1]
```diff
@@ -8,18 +8,25 @@
     "launch": "ssf-electron ./src/app.json",
     "start": "npm-run-all --parallel launch serve",
     "postinstall": "rimraf src && copyfiles -f node_modules/ssf-desktop-api-specification/src/** src",
-    "test": "echo \"No test specified\" && exit 0"
+    "test": "echo \"No test specified\" && exit 0",
+    "mocha": "mocha test/init.js"
```
Comment (AndyMJones) [inline packages/api-electron-demo/package.json]: Why don't we just have npm test run mocha?

File: packages/api-electron-demo/test/init.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  host: '127.0.0.1',
+  root: 'src',
+  open: false,
+  ignore: '*'
+};
+
+liveServer.start(params);
+
+describe('application launch', function() {
+  this.timeout(10000);
```
Comment (AndyMJones) [inline packages/api-electron-demo/test/init.js]: Why do we want a 10 second timeout?

File: packages/api-electron/main.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ let win;
 
 function createWindow() {
   // the appJson location is passed to the ssf-electron bin script
-  const appJsonPath = process.cwd() + '/' + process.argv[5];
+  let configLocation = process.env.TEST ? 'src/app.json' : process.argv[5];
```
Comment (AndyMJones) [inline packages/api-electron/main.js]: const

File: packages/api-openfin-demo/package.json [hunk 4]
```diff
@@ -8,19 +8,30 @@
     "launch": "openfin --launch --config ./src/app.json",
     "start": "npm-run-all --parallel serve launch",
     "postinstall": "rimraf src && copyfiles -f node_modules/ssf-desktop-api-specification/src/** src && copyfiles -f node_modules/ssf-desktop-api-openfin/dist/** src",
-    "test": "echo \"No test specified\" && exit 0"
+    "test": "echo \"No test specified\" && exit 0",
```
Comment (AndyMJones) [inline packages/api-openfin-demo/package.json]: ditto

File: packages/api-openfin-demo/test/init.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+  open: false,
+  ignore: '*'
+};
+
+liveServer.start(params);
+
+describe('application launch', function() {
+  let app;
+  let client;
+  let timeout = 30000;
```
Comment (AndyMJones) [inline packages/api-openfin-demo/test/init.js]: const

File: packages/api-electron-demo/package.json (hunk 1 above)
Comment (AndyMJones) [reply packages/api-electron-demo/package.json]: Read the description. Guessing its so it doesn't run on CI?

Comment (BenLambertNcl) [reply packages/api-electron-demo/package.json]: Exactly :)

File: packages/api-electron-demo/test/init.js (hunk 2 above)
Comment (BenLambertNcl) [reply packages/api-electron-demo/test/init.js]: We need to wait for electron/openfin to start before the tests start to run. The default is 2 seconds which is too short. The OpenFin tests acutally have a timeout of 30 seconds, so ill make this the same (will be needed once the tests are running on CI as they will run much slower.

**Review by AndyMJones — APPROVED** [review]

**Comment by BenLambertNcl** [comment]
Updated this PR with the final version of Electron and Openfin tests.

**Review by AndyMJones — APPROVED** [review]

**Comment by BenLambertNcl** [comment]
Closed by #70

**Pull request closed without merging**